libs/partners/pinecone/langchain_pinecone/embeddings.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+from typing import Dict, Iterable, List, Optional
+
+import aiohttp
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import (
+    BaseModel,
+    Extra,
+    Field,
+    SecretStr,
+    root_validator,
+)
+from langchain_core.utils import convert_to_secret_str
+from pinecone import Pinecone as PineconeClient  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class PineconeEmbeddings(BaseModel, Embeddings):
```

Comment:
should we mark this `@beta` while this in preview?

Reply:
I'll add